File: RentalApp_FASTAPI/api/services/brand_system_service.py
# api/services/brand_system_service.py

from api.repositories.brand_system_repository import BrandSystemRepository
from shared.schemas.brand_system_schema import BrandSystemCreate, BrandSystemUpdate, BrandSystemOut

class BrandSystemService:
    """Сервис-фасад для управления логикой Систем Бренда."""
    def __init__(self, repo: BrandSystemRepository):
        self.repo = repo

    async def get_all_paginated(self, skip: int, limit: int):
        """Получить все системы с пагинацией."""
        systems, total = await self.repo.get_all_paginated(skip, limit)
        # Преобразуем в схемы с правильными equipment_ids
        systems_out = [BrandSystemOut.from_orm_with_equipment_ids(system) for system in systems]
        return systems_out, total

    async def create(self, data: BrandSystemCreate):
        """Создать новую систему."""
        new_system = await self.repo.create_with_equipment(data)
        await self.repo.save()
        return BrandSystemOut.from_orm_with_equipment_ids(new_system)

    async def update(self, system_id: int, data: BrandSystemUpdate):
        """Обновить существующую систему."""
        updated_system = await self.repo.update_with_equipment(system_id, data)
        await self.repo.save()
        return BrandSystemOut.from_orm_with_equipment_ids(updated_system)
        
    async def delete(self, system_id: int):
        """Удалить систему."""
        await self.repo.delete_by_id(system_id)
        await self.repo.save()

File: rental-app-main/src/core/services/BrandSystemService.js
// src/core/services/BrandSystemService.ts
import { api } from "@/lib/api";
export class BrandSystemService {
    static async getAll() {
        const response = await api.get("/admin/brand-systems/");
        return response.data;
    }
    static async create(data) {
        const response = await api.post("/admin/brand-systems/", data);
        return response.data;
    }
    static async update({ id, data }) {
        const response = await api.put(`/admin/brand-systems/${id}`, data);
        return response.data;
    }
    static async delete(id) {
        await api.delete(`/admin/brand-systems/${id}`);
    }
}
